src/vaults/yields.utils.ts

```diff
@@ -421,12 +423,22 @@ async function evaluateYieldEvents(chain: Chain, vault: VaultDefinitionModel): P
   // this error should bubble up to yield source persistence
   // from a practical perspective this is here to allow vaults to retain calculated yield if there is no history
   // this is primary needed for a clean resync of harvest data without impacting 'synced' data
-  if (yieldEvents.length === 0) {
-    throw new Error(`${vault.name} has no recent harvests, it is either not synced, or effectively deprecated`);
+  if (vault.state !== VaultState.Discontinued && yieldEvents.length === 0) {
+    throw new Error(`${vault.name} ${NO_HARVESTS_MESSAGE}`);
   }
 
-  const relevantYieldEvents = filterPerformanceItems(vault, yieldEvents).sort((a, b) => a.timestamp - b.timestamp);
   const tokenEmissionAprs: VaultEmissionData = new Map();
+
+  if (vault.state === VaultState.Discontinued) {
```

Comment:
handle discontinued vaults early - do not hit errors with trying to parse (highly likely) no events - just return we know no yields (FROM HARVESTS)